mkt/langpacks/models.py

```diff
@@ -0,0 +1,108 @@
+ # -*- coding: utf-8 -*-
+import hashlib
+import os.path
+
+from django.conf import settings
+from django.core.files.storage import default_storage as storage
+from django.db import models
+from django.utils.functional import lazy
+
+import commonware.log
+from uuidfield.fields import UUIDField
+
+from mkt.files.models import nfd_str
+from mkt.langpacks.utils import LanguagePackParser
+from mkt.site.models import ModelBase
+from mkt.site.storage_utils import copy_stored_file
+from mkt.site.utils import smart_path
+
+
+log = commonware.log.getLogger('z.versions')
+
+
+LANGUAGE_CHOICES = lazy(lambda l: dict(l).items(), list)(settings.LANGUAGES)
+
+
+class LangPack(ModelBase):
+    # Primary key is a uuid in order to be able to set it in advance (we need
+    # something unique for the filename, and we don't have a slug).
+    uuid = UUIDField(primary_key=True, auto=True)
+
+    # Fields for which the manifest is the source of truth - can't be
+    # overridden by the API.
+    language = models.CharField(choices=LANGUAGE_CHOICES,
+                                default=settings.LANGUAGE_CODE, max_length=10)
```

Comment:
Right now they don't, because we bypass the choices, but yeah, that's the plan. Going to add it.